### DISCUSSION TITLE
Git cliff with a single tag and `--latest` option

### DISCUSSION THREAD

**Discussion by rarescosma** [orhun/git-cliff#893] [Q&A] [answered]
Hi,

I'm using `git cliff --latest` to obtain a diff/changelog between the current (latest) tag and the previous.

However, when the repository only has one tag, the root commit is not included in the diff.

So `git cliff` produces the correct output: 

```
# git cliff

## 1.0.0 - 2024-09-28

### Changelog

- spawning overlords [#6b45ca4](commit/6b45ca47f59f7f3580aae38e1852ae83f90fd0ba)
- down to 50kb and learned a whole bunch about rust [#519e55a](commit/519e55a6bcd24372502ac3588e19606c81ca79ac)
- moar just targets [#4b083e2](commit/4b083e2195f6f47a966efe7df77ebd5ea1de34a2)
- go to lexopt instead of pointer shenanigans [#9649c8e](commit/9649c8ee90b1b85f7ef0d216e1973463138bb102)
...
```

Whereas `git cliff --latest` produces:

```
# git cliff --latest

## 1.0.0 - 2024-09-28

### Changelog

- down to 50kb and learned a whole bunch about rust [#519e55a](commit/519e55a6bcd24372502ac3588e19606c81ca79ac)
- moar just targets [#4b083e2](commit/4b083e2195f6f47a966efe7df77ebd5ea1de34a2)
- go to lexopt instead of pointer shenanigans [#9649c8e](commit/9649c8ee90b1b85f7ef0d216e1973463138bb102)
...
```

Is this the expected behavior?

Thanks!

**Comment by orhun** [comment]
Hello 👋🏼 

That looks like the expected behavior to me. There isn't any previous tag other than the first one, so the first one is being used. (Or do you mean the missing commit?)

Were you expecting to see a different output?

**Reply by rarescosma** [reply]
Hello 👋🏼 

Yes, I was expecting to see the root commit (first commit) in the repo included, regardless of using `--latest` or not, when there is only 1 tag.

Otherwise I have to add it manually on each first release of a project.

`--latest` works fine with 2 tags or more.

Thanks!

**Reply by rarescosma** [reply]
Hi, I think I can express this better in code, so have opened a PoC pull request about it: https://github.com/orhun/git-cliff/pull/901

**Reply by orhun** [reply, marked as answer]
Thanks for creating the PR! Taking a look at it now
